modules/services/repositories/src/debug/kotlin/com/android/billingclient/api/ProductDetailsModule.kt

```diff
@@ -0,0 +1,163 @@
+package com.android.billingclient.api
+
+import au.com.shiftyjelly.pocketcasts.repositories.subscription.ProductDetailsInterceptor
+import com.android.billingclient.api.BillingClient.BillingResponseCode.OK
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+
+@Module
+@InstallIn(SingletonComponent::class)
+object ProductDetailsModule {
+    @Provides
+    fun provideProductDetailsInterceptor(): ProductDetailsInterceptor = MockingProductDetailsInterceptor()
+}
+
+private class MockingProductDetailsInterceptor : ProductDetailsInterceptor {
```

Comment:
Thanks for making this change. It's going to make it so much easier to develop and test those areas of the app.

I had some minor comments, feel free to ignore:

1. Should we move the `MockingProductDetailsInterceptor` class into the `main` folder and only include the DI code in `debug` and `debugProd` rather than duplicate this code twice? 
2. Should we use our package name and not Google's `com.android`? It feels a bit wrong using theirs.

Reply:
1. I don't want to risk it getting into production code. If you feel this is not an issue feel free to move it.
2. This won't work. Constructor for `ProductDetails` is package private;